platform/ios/test/MGLAnnotationViewTests.m

```diff
@@ -0,0 +1,62 @@
+#import <Mapbox/Mapbox.h>
+#import <XCTest/XCTest.h>
+
+static NSString * const MGLTestAnnotationReuseIdentifer = @"MGLTestAnnotationReuseIdentifer";
+
+@interface MGLTestAnnotation : NSObject <MGLAnnotation>
+@property (nonatomic, assign) CLLocationCoordinate2D coordinate;
+@end
+
+@implementation MGLTestAnnotation
+@end
+
+@interface MGLAnnotationViewTests : XCTestCase <MGLMapViewDelegate>
+@property (nonatomic) XCTestExpectation *expectation;
+@property (nonatomic) MGLMapView *mapView;
+@end
+
+@implementation MGLAnnotationViewTests
+
+- (void)testAnnotationView
+{
+    _mapView = [[MGLMapView alloc] initWithFrame:CGRectZero];
+    _mapView.delegate = self;
+    
+    _expectation = [self expectationWithDescription:@"test annotation property"];
+    
+    MGLTestAnnotation *annotation = [MGLTestAnnotation new];
+    [_mapView addAnnotation:annotation];
+    
+    [self waitForExpectationsWithTimeout:1 handler:nil];
+}
+
+- (void)performTestWithAnnotationView:(MGLAnnotationView *)annotationView
```

Comment:
The flow in this test class is a bit hard to reason about. I'm adding XCUITests that would make this sort of thing much easier in #4837. In the meantime, I think a better structure for this test would be to perform these checks at the end of -testAnnotationView, after the wait. -mapView:viewForAnnotation: should fulfill the expectation that the delegate callback gets called, allowing -testAnnotationView to continue running.